tests/envs/utils.py

```diff
@@ -0,0 +1,78 @@
+"""Finds all the specs that we can test with"""
+from typing import Optional
+
+import numpy as np
+
+import gym
+from gym import logger
+from gym.envs.registration import EnvSpec
+
+
+def try_make_env(env_spec: EnvSpec) -> Optional[gym.Env]:
+    """Tries to make the environment showing if it is possible. Warning no wrappers, including time limit."""
+    try:
+        return env_spec.make(disable_env_checker=True).unwrapped
+    except ImportError as e:
+        logger.warn(f"Not testing {env_spec.id} due to error: {e}")
+        return None
+
+
+# Tries to make all environment to previous recreation if not necessary.
+all_testing_initialised_envs = list(
+    filter(None, [try_make_env(env_spec) for env_spec in gym.envs.registry.values()])
+)
+
+# All testing, mujoco and gym environment specs
+all_testing_env_specs = [env.spec for env in all_testing_initialised_envs]
```

Comment:
So you first get all the env specs -> initialize the envs -> discard erroneous initializations -> get the specs again from remaining envs -> use them in tests where they are initialized again. 

There's some duplication here, instead maybe you could directly pass the initialized filtered envs to the tests?

Reply:
Yes, that is basically how it works. 
I have tried to use the pre-initialised environments but will check again

Reply:
Although if we ignore the duplication, functionally this should be fine. There would probably be unnecessary complications in using the same pre-initialized envs everywhere (which if we want to avoid we have to create deep copies, and even then there might be issues). So, thinking again, maybe it's fine unless there's a more elegant solution